### ISSUE TITLE
gccd doesn't work if no output specified

### ISSUE THREAD

**Issue by mark2185** [ohmyzsh/ohmyzsh#12023]
### Describe the bug

According to the comments in the `gccd` function, `gccd git@github.com:ohmyzsh/ohmyzsh.git` should work, but I get this:
```shell
Cloning into 'ohmyzsh'...
remote: Enumerating objects: 31651, done.
remote: Counting objects: 100% (2566/2566), done.
remote: Compressing objects: 100% (307/307), done.
remote: Total 31651 (delta 2429), reused 2327 (delta 2259), pack-reused 29085
Receiving objects: 100% (31651/31651), 10.30 MiB | 9.36 MiB/s, done.
Resolving deltas: 100% (16526/16526), done.
gccd:cd:9: no such file or directory: local
```

I'm pretty sure it was introduced in [this commit](https://github.com/ohmyzsh/ohmyzsh/commit/ff6dac56a256e376a10b19a4948ada7df4509c71).

### Steps to reproduce

1. Enable plugin `git`
2. Run `gccd git@github.com:ohmyzsh/ohmyzsh.git`

### Expected behavior

I expect to be positioned in the cloned repository.

### OS / Linux distribution

Linux lapitopi 6.5.9-arch2-1 #1 SMP PREEMPT_DYNAMIC Thu, 26 Oct 2023 00:52:20 +0000 x86_64 GNU/Linux

### Zsh version

5.9

### Terminal emulator

Kitty, guake

### If using WSL on Windows, which version of WSL

None

### Additional context

I tried debugging it, but I'm not very well versed in the cipher called `zsh` so I'm opening up an issue instead.

**Comment by mcornella** [comment]
Hi Luka, thanks for the report. I can't actually reproduce this, could you run the same command again with trace enabled and post the output? 
```sh
functions -t gccd
gccd git@github.com:ohmyzsh/ohmyzsh.git
```

**Comment by mark2185** [comment]
Here's the output
```
 $> gccd git@github.com:ohmyzsh/ohmyzsh.git
+gccd:2> local repo=local
+gccd:5> git clone --recurse-submodules git@github.com:ohmyzsh/ohmyzsh.git
Cloning into 'ohmyzsh'...
remote: Enumerating objects: 31651, done.
remote: Counting objects: 100% (2566/2566), done.
remote: Compressing objects: 100% (302/302), done.
remote: Total 31651 (delta 2429), reused 2332 (delta 2264), pack-reused 29085
Receiving objects: 100% (31651/31651), 10.30 MiB | 6.59 MiB/s, done.
Resolving deltas: 100% (16526/16526), done.
+gccd:9> [[ -d git@github.com:ohmyzsh/ohmyzsh.git ]]
+gccd:9> cd local
gccd:cd:9: no such file or directory: local
```

And here's with an additional argument
```
$> gccd git@github.com:ohmyzsh/ohmyzsh.git asdffff
+gccd:2> local repo=local
+gccd:5> git clone --recurse-submodules git@github.com:ohmyzsh/ohmyzsh.git asdffff
Cloning into 'asdffff'...
remote: Enumerating objects: 31651, done.
remote: Counting objects: 100% (2570/2570), done.
remote: Compressing objects: 100% (309/309), done.
remote: Total 31651 (delta 2431), reused 2329 (delta 2261), pack-reused 29081
Receiving objects: 100% (31651/31651), 10.30 MiB | 8.33 MiB/s, done.
Resolving deltas: 100% (16526/16526), done.
+gccd:9> [[ -d asdffff ]]
+gccd:9> cd asdffff
+_ksi_report_pwd:0> print -nu 13 '\e]7;kitty-shell-cwd://lapitopi/tmp/asdffff\a'
```

**Comment by carlosala** [comment]
@mcornella indeed, I can reproduce. The difference is `setopt extendedglob`.
https://asciinema.org/a/619368

**Comment by mcornella** [comment]
Indeed that was the issue. Fixed in b6bb133f.
